flowmachine/tests/server/test_query_proxy.py

```diff
@@ -168,10 +176,60 @@ def dummy_construct_query_object(query_kind, params):
     #
     query_proxy.run_query_async()
     q.store.assert_called_once_with()
-    assert "done" == query_proxy.poll()
+    assert "completed" == query_proxy.poll()
 
     #
     # Get SQL code and check it is as expected
     #
     sql = query_proxy.get_sql()
     assert "SELECT * FROM dummy_table" == sql
+
+
+@pytest.mark.parametrize(
+    "current_state, expected_error",
+    [
+        (QueryState.KNOWN, MissingQueryError),
```

Comment:
Similar to my comment above, I'm wondering whether it's worth always raising the same error. I'll keep that in mind for my re-working / removal of the QueryProxy class.